### ISSUE TITLE
Treat creates with "duplicateWarning" and "idNotUnique" responses

### ISSUE THREAD

**Issue by mconf-daileon** [mconf/bigbluebutton_rails#83]
---

Author Name: **Leonardo Daronco** (Leonardo Daronco)
Original Redmine Issue: 606, http://dev.mconf.org/redmine/issues/606
Original Date: 2013-02-19
Original Assignee: Alister Machado

---

* @duplicateWarning@ happens when you call a @create@ using the same *meetingID, moderator password and attendee password* of a meeting that is already created. Since a @create@ is only called when the meeting is not running, this can only happen if the application can't see the meeting that is already created (if using a load balancer that filters the meetings the application can see, for example). It's rare, but should be treated anyways. It's already being done in https://github.com/mconf/bigbluebutton_rails/blob/master/app/models/bigbluebutton_room.rb#L126, but should be reviewed.

* @idNotUnique@ happens when you call a create using the same *meetingID* of a meeting that is already created but using a *different moderator password or attendee password*. This is a lot more likely to happen than @duplicateWarning@ (since @create@ is only called if the meeting is not running), but it is not being treated yet.

If any of the errors occur the gem should randomize another meetingID and try again.

And maybe there are other errors similar to @idNotUnique@ that should also be treated equally.

**Comment by mconf-daileon** [comment]
---

Original Redmine Comment
Author Name: **Leonardo Daronco** (Leonardo Daronco)
Original Date: 2013-06-12T18:54:23Z

---

This has been discussed in BigBlueButton's mailing list and the idea has changed a lot from the original idea described in this issue. See https://groups.google.com/forum/#!msg/bigbluebutton-dev/Cz9zcPbRZfI/CLX1IdQgA9AJ

The current version of @BigbluebuttonRails@ uses unique global meetingIDs for conferences and so the generation of random meetingIDs was removed.
With static meetingIDs the problem we have is what is described in this issue: https://code.google.com/p/bigbluebutton/issues/detail?id=1324
In short, a meeting that finish can stay in memory in the BigBlueButton server for a while, and while it is there the user won't be able to create the meeting again with, for example, different parameters to set the record flag.

The proposed solution is:
* Check if the meeting is running. If it is, we don't need to create the meeting again.
* If it is not running, call 'end' to finish the old meeting and only then create it again.

This has to be implemented and tested against a real server.

**Comment by mconf-daileon** [comment]
---

Original Redmine Comment
Author Name: **Leonardo Daronco** (Leonardo Daronco)
Original Date: 2014-08-12T16:18:39Z

---

Checking if a meeting is running or not is not always ideal. A meeting can be in memory but not running if there's no one in it. A better solution would be to find if the meeting is in memory or not.

So the gem should either:

1. Call @end@ always in case the meeting is not running (as proposed in the comment above); or
1. Instead of calling @isMeetingRunning@ to check if the meeting is running, the gem could call @getMeetingInfo@. It would also have the @running@ flag in it, but would also indicate whether the meeting is created (in memory) or not. If it is in memory (running or not), the gem should call an @end@ first. Otherwise it doesn't need to call @end@.
